src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -50,6 +56,39 @@ public function __construct(?string $name, NodeInterface $parent = null, string
         $this->pathSeparator = $pathSeparator;
     }
 
+    /**
+     * Register possible (dummy) values for a dynamic placeholder value. Matching configuration values will be processed
+     * with a provided value, one by one. After a provided value is successfully processed the configuration value is
+     * returned as is, thus preserving the placeholder.
+     */
+    public static function setPlaceholder(string $placeholder, array $values): void
+    {
+        if (!$values) {
+            throw new \InvalidArgumentException('At least one value must be provided.');
+        }
+
+        self::$placeholders[$placeholder] = $values;
+    }
+
+    /**
+     * Set a common prefix for dynamic placeholder values. Matching configuration values will be skipped from being
```

Comment:
Sets + same comment as above about separating the first sentence from the other ones.